package/MDAnalysis/lib/distances.py

```diff
@@ -608,6 +609,61 @@ def _pkdtree_capped(reference, configuration, max_cutoff, min_cutoff=None, box=N
                 distances.append(dist[num])
     return pairs, distances
 
+def _nsgrid_capped(reference, configuration, max_cutoff, min_cutoff=None,
+                   box=None):
+    """Search all the pairs in *reference* and *configuration* within 
+    a specified distance using Grid Search
+
+    
+    Parameters
+    -----------
+    reference : array
+        reference coordinates array with shape ``reference.shape = (3,)``
+        or ``reference.shape = (len(reference), 3)``.
+    configuration : array
+        Configuration coordinate array with shape ``reference.shape = (3,)``
+        or ``reference.shape = (len(reference), 3)``
+    max_cutoff : float
+        Maximum cutoff distance between the reference and configuration
+    min_cutoff : (optional) float
+        Minimum cutoff distance between reference and configuration [None]
+    box :  array 
+        The dimensions, if provided, must be provided in the same
+        The unitcell dimesions for this system format as returned
+        by :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`:
+        ``[lx,ly, lz, alpha, beta, gamma]``. Minimum image convention
+        is applied if the box is provided 
+
+    Note
+    ----
+    Non Periodic Boundary conditions can-not be handled currently
+
+    """
+    from .nsgrid import FastNS
+    if reference.shape == (3, ):
+        reference = reference[None, :]
+    if configuration.shape == (3, ):
+        configuration = configuration[None, :]
+    all_coords = np.concatenate([configuration, reference])
+    mapping = np.arange(len(reference), dtype=np.int64)
+    gridsearch = FastNS(box, max_cutoff, all_coords)
+    gridsearch.prepare()
+    search_ids = np.arange(len(configuration), len(all_coords))
+    results = gridsearch.search(search_ids=search_ids)
+    pairs = results.get_pairs()
+    pairs[:, 1] = undo_augment(pairs[:, 1], mapping, len(configuration))
```

Comment:
Why are we doing an undo_augment if we never augmented?   Is this because we had to concatenate the coordinates together?

Reply:
Yeah, I understand it could become confusing. But I just used it since we already had that function which served the purpose. But yes, since we are concatenating and to get the same results from other capped_functions, all the indices should be translated back to their corresponding group indices.